nntrainer/layers/multi_head_attention_layer.cpp

```diff
@@ -184,73 +195,77 @@ void MultiHeadAttentionLayer::finalize(InitLayerContext &context) {
+                                tensor_type);
     weight_idx[AttentionParams::query_fc_bias] = context.requestWeight(
       query_fc_bias_dim, weight_initializer, weight_regularizer,
       weight_regularizer_constant, weight_decay, "query_fc_bias", true);
   }
 
   /** weight/bias for key fc */
   TensorDim key_fc_weight_dim(
-    {1, 1, key_width, num_heads * projected_key_dim_prop});
+    {1, 1, key_width, num_heads * projected_key_dim_prop}, tensor_type);
   weight_idx[AttentionParams::key_fc_weight] = context.requestWeight(
     key_fc_weight_dim, weight_initializer, weight_regularizer,
     weight_regularizer_constant, weight_decay, "key_fc_weight", true);
   if (!disable_bias) {
-    TensorDim key_fc_bias_dim({1, 1, 1, num_heads * projected_key_dim_prop});
+    TensorDim key_fc_bias_dim({1, 1, 1, num_heads * projected_key_dim_prop},
+                              tensor_type);
     weight_idx[AttentionParams::key_fc_bias] = context.requestWeight(
       key_fc_bias_dim, weight_initializer, weight_regularizer,
       weight_regularizer_constant, weight_decay, "key_fc_bias", true);
   }
 
   /** weight/bias for value fc */
   TensorDim value_fc_weight_dim(
-    {1, 1, value_width, num_heads * projected_value_dim_prop});
+    {1, 1, value_width, num_heads * projected_value_dim_prop}, tensor_type);
   weight_idx[AttentionParams::value_fc_weight] = context.requestWeight(
     value_fc_weight_dim, weight_initializer, weight_regularizer,
     weight_regularizer_constant, weight_decay, "value_fc_weight", true);
   if (!disable_bias) {
-    TensorDim value_fc_bias_dim(
-      {1, 1, 1, num_heads * projected_value_dim_prop});
+    TensorDim value_fc_bias_dim({1, 1, 1, num_heads * projected_value_dim_prop},
+                                tensor_type);
     weight_idx[AttentionParams::value_fc_bias] = context.requestWeight(
       value_fc_bias_dim, weight_initializer, weight_regularizer,
       weight_regularizer_constant, weight_decay, "value_fc_bias", true);
   }
 
   /** weight/bias for out fc */
   TensorDim fc_weight_dim(
-    {1, 1, num_heads * projected_value_dim_prop, output_shape});
+    {1, 1, num_heads * projected_value_dim_prop, output_shape}, tensor_type);
   weight_idx[AttentionParams::fc_weight] = context.requestWeight(
     fc_weight_dim, weight_initializer, weight_regularizer,
     weight_regularizer_constant, weight_decay, "fc_weight", true);
   if (!disable_bias) {
-    TensorDim fc_bias_dim({1, 1, 1, output_shape});
+    TensorDim fc_bias_dim({1, 1, 1, output_shape}, tensor_type);
     weight_idx[AttentionParams::fc_bias] = context.requestWeight(
       fc_bias_dim, weight_initializer, weight_regularizer,
       weight_regularizer_constant, weight_decay, "fc_bias", true);
   }
 
   /** tensor for output of query fc */
   TensorDim projected_query_dim(
-    {batch_size, 1, query_height, num_heads * projected_query_dim_prop});
+    {batch_size, 1, query_height, num_heads * projected_query_dim_prop},
+    tensor_type);
```

Comment:
I'm not sure if this is tensor_type or {context.getFormat(), context.getWeightDataType()}. plz check once again.